Day-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,68 @@
-import React from 'react';
+import React, { Component }  from 'react';
 import './App.css';
+import CalculatorDisplay from './components/DisplayComponents/CalculatorDisplay';
+import NumberButton from './components/ButtonComponents/NumberButton';
+import ActionButton from './components/ButtonComponents/ActionButton';
 
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
+class App extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      result: 0,
+    }
+
+    this.numClicked = this.numClicked.bind(this);
+  }
+
+  numClicked(result) {
+    let newCalc = '';
+    newCalc += (this.state.result === 0 ? '' : this.state.result);
+    newCalc += result;
+
+    if (result ==='clear') newCalc = 0;
+
+    this.setState({result: newCalc});
+  }
+
+  render() {
+    const wideButton = {
+      width: '339px', 
+      color: 'black', 
+      background: 'white'
+    };
+
+    const rows = [
+      ['clear', '÷'],
+      ['7', '8', '9', 'x'],
+      ['4', '5', '6', '-'],
+      ['1', '2', '3', '+'],
+      ['0', '=']
+    ]
+
+    return (
+      <div className='calculator-container'>
+        <CalculatorDisplay result={this.state.result}/>
+        {
+          rows.map((row, i) => {
```

Comment:
It's best to have the child component do the mapping, otherwise it defeat the purpose of reusability.